ball_detection.py

```diff
@@ -0,0 +1,62 @@
+import cv2
+import numpy as np
+import imutils
+import math
+
+KNOWN_DIAMETER_IN = 9.5  # in
+FOCAL_LENGTH = 1367.043233  # 94% accuracy
+RADIUS_THRESH = 85
+MIN_BORDERS = 5
+FOV = 78
+DIM = 1920
+
+
+class BallDetection:
+
+    def __init__(self, frame):
+        self.frame = frame
+        self.x_val = int(frame.shape[1])
+        self.y_val = int(frame.shape[0])
+
+    def detect_ball(self):
+        distance = -1
+        area = -1
+        angle = -1
+        frame = cv2.cvtColor(self.frame, cv2.COLOR_BGR2GRAY)
+        self.prep_frame()
+        cnts = cv2.findContours(frame, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_NONE)
+        cnts = imutils.grab_contours(cnts)
+
+        if len(cnts) > 0:
+            for cnt in cnts:
+                ((x, y), radius) = cv2.minEnclosingCircle(cnt)
+                if radius >= RADIUS_THRESH:
+                    perimeter = cv2.arcLength(cnt, closed=True)
+                    borders = cv2.approxPolyDP(curve=cnt, epsilon=0.0085 * perimeter, closed=True)
+                    if len(borders) >= MIN_BORDERS:
+                        if cv2.contourArea(cnt) > area:  # finding the closest ball
```

Comment:
In fact, I wonder if we can just use the area threshold and skip the circumference check entirely.

Reply:
yea, I just verified using the area threshold and it works the same